core/lib/libtomcrypt/src/tee_ltc_provider.c

```diff
@@ -400,6 +401,39 @@ TEE_Result crypto_hash_get_ctx_size(uint32_t algo, size_t *size)
 	return TEE_SUCCESS;
 }
 
+TEE_Result crypto_hash_alloc_ctx(void **ctx_ret, uint32_t algo)
+{
+	TEE_Result res;
+	size_t ctx_size;
+	void *ctx;
+
+	res = hash_get_ctx_size(algo, &ctx_size);
+	if (res)
+		return res;
+
+	ctx = calloc(1, ctx_size);
+	if (!ctx)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	*ctx_ret = ctx;
+	return TEE_SUCCESS;
+}
+
+void crypto_hash_free_ctx(void *ctx, uint32_t algo __unused)
+{
+	free(ctx);
```

Comment:
I think here should add ctx check before free it. If crypto_hash_alloc_ctx() failed, it will go to err or out and crypto_hash_free_ctx(). ctx may be still NULL. 
Or we can add ctx check when call crypto_hash_free_ctx().

Reply:
Calling `free()` with a `NULL` pointer argument is valid in C.